packages/zql/src/query/query-registry.ts

```diff
@@ -327,26 +330,29 @@ function createCustomQueryBuilder<
   T extends keyof S['tables'] & string,
   R,
   C,
-  Args extends ReadonlyJSONValue | undefined,
+  ArgsInput extends ReadonlyJSONValue | undefined,
+  ArgsOutput extends ReadonlyJSONValue | undefined,
   HasArgs extends boolean,
 >(
-  queryDef: QueryDefinition<S, T, R, C, any, Args>,
+  queryDef: QueryDefinition<S, T, R, C, ArgsInput, ArgsOutput>,
   name: string,
-  args: Args,
+  inputArgs: ArgsInput,
+  validatedArgs: ArgsOutput,
   hasArgs: HasArgs,
-): CustomQuery<S, T, R, C, Args, HasArgs> {
+): CustomQuery<S, T, R, C, ArgsInput, ArgsOutput, HasArgs> {
   const {validator} = queryDef;
 
   // The callable function that sets args
-  const builder = (args: Args) => {
+  const builder = (args: ArgsInput) => {
     if (hasArgs) {
       throw new Error('args already set');
     }
-    const validatedArgs = validateInput(name, args, validator, 'query');
-    return createCustomQueryBuilder<S, T, R, C, Args, true>(
+    const validated = validateInput(name, args, validator, 'query');
+    return createCustomQueryBuilder<S, T, R, C, ArgsInput, ArgsOutput, true>(
       queryDef,
       name,
-      validatedArgs,
+      args,
+      validated,
       true,
     );
   };
```

Comment:
Aha! This was the bug. This must have been hard to debug. 

When it changes the type of the args it didn't update the Args type parameter.

The runtime behavior was correct but the type was not.

Thanks.